src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -298,4 +301,77 @@ private function getPrecisionFromCart(Cart $cart): int
 
         return $computingPrecision->getPrecision((int) $currency->precision);
     }
+
+    /**
+     * Update order details after a specific price has been created or updated
+     *
+     * @param Order $order
+     * @param OrderDetail $updatedOrderDetail
+     * @param Product $product
+     * @param int|null $combinationId
+     * @param int $productQuantity
+     * @param Number $priceTaxIncluded
+     * @param Number $priceTaxExcluded
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function updateOrderDetailsWithSameProduct(
+        Order $order,
+        OrderDetail $updatedOrderDetail,
+        Product $product,
+        ?int $combinationId,
+        Number $priceTaxIncluded,
+        Number $priceTaxExcluded,
+        int $computingPrecision
+    ): void {
+        foreach ($order->getOrderDetailList() as $row) {
+            $orderDetail = new OrderDetail($row['id_order_detail']);
+            if ((int) $orderDetail->product_id !== (int) $product->id) {
+                continue;
+            }
+            if (!empty($combinationId) && (int) $combinationId !== (int) $orderDetail->product_attribute_id) {
+                continue;
+            }
+            if ($updatedOrderDetail->id == $orderDetail->id) {
+                continue;
+            }
+            $orderDetail->unit_price_tax_excl = (float) (string) $priceTaxExcluded;
+            $orderDetail->unit_price_tax_incl = (float) (string) $priceTaxIncluded;
+            $orderDetail->total_price_tax_excl = Tools::ps_round((float) (string) $priceTaxExcluded * $orderDetail->product_quantity, $computingPrecision);
+            $orderDetail->total_price_tax_incl = Tools::ps_round((float) (string) $priceTaxIncluded * $orderDetail->product_quantity, $computingPrecision);
+
+            $orderDetail->update();
+        }
+    }
+
+    /**
+     * This method takes care of multi-invoices, all invoices are updated
+     *
+     * @param Order $order
+     * @param int $precision
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function updateOrderInvoices(Order $order, int $precision): void
+    {
+        $orderInvoices = OrderInvoice::getInvoicesByOrderId($order->id);
+
+        foreach ($orderInvoices as $invoice) {
```

Comment:
You should never compute order totals yourself There are too many business rules (cart rules especially) that can be applied The only one that can do this correct is the Cart itself (or indirectly the ) by using `$cart->getOrderTotal`

Like it's been done in the `updateOrderInvoice` function, it's actually this method that you should modify, so that it doesn't update only one invoice any more but all of them Be careful as well about shipping, it should only be computed for the first invoice